src/app/pages/send/send-crypto-asset-form/form/btc/use-generate-bitcoin-raw-tx.ts

```diff
@@ -4,45 +4,47 @@ import { hexToBytes } from '@stacks/common';
 import * as btc from 'micro-btc-signer';
 
 import { getBtcSignerLibNetworkByMode } from '@shared/crypto/bitcoin/bitcoin.network';
-import { logger } from '@shared/logger';
 import { BitcoinSendFormValues } from '@shared/models/form.model';
 
 import { btcToSat } from '@app/common/money/unit-conversion';
 import { useGetUtxosByAddressQuery } from '@app/query/bitcoin/address/utxos-by-address.query';
-import { useBitcoinFeeRatesInVbytes } from '@app/query/bitcoin/fees/fee-estimates.hooks';
+import { useBitcoinFeeRate } from '@app/query/bitcoin/fees/fee-estimates.hooks';
 import {
   useCurrentBitcoinAddressIndexKeychain,
   useCurrentBtcAccountAddressIndexZero,
   useSignBitcoinTx,
 } from '@app/store/accounts/blockchain/bitcoin/bitcoin-account.hooks';
 import { useCurrentNetwork } from '@app/store/networks/networks.selectors';
 
-import { coinselect } from '../../family/bitcoin/coinselect/coinselect';
+import { determineUtxosForSpend } from '../../family/bitcoin/coinselect/local-coin-selection';
 
-export function useGenerateBitcoinRawTx() {
+export function useGenerateSignedBitcoinTx() {
   const currentAccountBtcAddress = useCurrentBtcAccountAddressIndexZero();
-  const utxos = useGetUtxosByAddressQuery(currentAccountBtcAddress).data;
+  const { data: utxos } = useGetUtxosByAddressQuery(currentAccountBtcAddress);
   const currentAddressIndexKeychain = useCurrentBitcoinAddressIndexKeychain();
   const signTx = useSignBitcoinTx();
   const network = useCurrentNetwork();
-  const { data: feeRate } = useBitcoinFeeRatesInVbytes();
+  const { data: feeRate } = useBitcoinFeeRate();
 
   return useCallback(
     (values: BitcoinSendFormValues) => {
       if (!utxos) return;
       if (!feeRate) return;
 
-      // console.log(utxos);
       const networkMode = getBtcSignerLibNetworkByMode(network.chain.bitcoin.network);
 
       try {
         const tx = new btc.Transaction();
 
-        const { inputs, outputs, fee } = coinselect(
+        const { inputs, outputs, fee } = determineUtxosForSpend({
           utxos,
-          [{ address: values.recipient, value: btcToSat(values.amount).toNumber() }],
-          feeRate.estimates[1].feeRate
-        );
+          recipient: values.recipient,
+          amount: btcToSat(values.amount).toNumber(),
+          feeRate: feeRate.fastestFee,
+        });
+
+        // eslint-disable-next-line no-console
+        console.log('coinselect', { inputs, outputs, fee });
```

Comment:
You meaning to keep these console.logs (on L81 too)?

Reply:
yeah can stay there for now, but we'll need to tidy our bedrooms as soon as it's launched